src/publicTypes.ts

```diff
@@ -5,19 +5,22 @@ import {
   TransactionRequest,
 } from './types';
 
-/**
- * SourceDefinition object
- * @typedef {Object} SourceDefinition
- * @property {string} [url] - main blockchain api url
- * @property {string | number} [network] - blockchain network
- */
-export interface SourceDefinition {
+export interface NamingServiceConfig {
   url?: string;
   network?: string | number;
   registry?: string;
+  proxyReaderAddress?: string;
   provider?: Provider;
+  api?: string;
 }
 
+export type SourceConfig = {
```

Comment:
@JohnnyJumper This structure does not comply with proposal https://github.com/unstoppabledomains/resolution/pull/115/files#diff-8483b709b167c8d30eb358b652ae5214044d40b0795aa626bffecf4bca091132R55

The main idea is to have separate type for each provider and strict alternatives.